src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -643,24 +647,25 @@ private Http2Stream GetStream<TContext>(IHttpApplication<TContext> application)
 
         private Http2StreamContext CreateHttp2StreamContext()
         {
-            return new Http2StreamContext
-            {
-                ConnectionId = ConnectionId,
-                StreamId = _incomingFrame.StreamId,
-                ServiceContext = _context.ServiceContext,
-                ConnectionFeatures = _context.ConnectionFeatures,
-                MemoryPool = _context.MemoryPool,
-                LocalEndPoint = _context.LocalEndPoint,
-                RemoteEndPoint = _context.RemoteEndPoint,
-                StreamLifetimeHandler = this,
-                ClientPeerSettings = _clientSettings,
-                ServerPeerSettings = _serverSettings,
-                FrameWriter = _frameWriter,
-                ConnectionInputFlowControl = _inputFlowControl,
-                ConnectionOutputFlowControl = _outputFlowControl,
-                TimeoutControl = TimeoutControl,
-                InitialExecutionContext = _context.InitialExecutionContext,
-            };
+            var streamContext = new Http2StreamContext(
+                ConnectionId,
+                protocols: default,
+                _context.ServiceContext,
+                _context.ConnectionFeatures,
+                _context.MemoryPool,
+                _context.LocalEndPoint,
+                _context.RemoteEndPoint,
+                _incomingFrame.StreamId,
+                streamLifetimeHandler: this,
+                _clientSettings,
+                _serverSettings,
+                _frameWriter,
+                _inputFlowControl,
+                _outputFlowControl);
+            streamContext.TimeoutControl = _context.TimeoutControl;
+            streamContext.InitialExecutionContext = _context.InitialExecutionContext;
```

Comment:
Any particular reason for not including both of these above properties in the context ctor?

Reply:
I made values that are settable as properties only.